Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -103,6 +103,105 @@ extension BSONValue where Self: Equatable {
+
+    /// Create a `Decimal128` from this `BSONNumber`.
+    /// This will return nil if the conversion cannot result in an exact representation.
+    func toDecimal128() -> Decimal128?
+}
+
+/// Default conformance to `BSONNumber` for `BinaryInteger`s.
+extension BSONNumber where Self: BinaryInteger {
+    /// Create an `Int` from this `BinaryInteger`.
+    /// This will return nil if the conversion cannot result in an exact representation.
+    public func toInt() -> Int? {
+        if let selfAsInt = self as? Int {
+            return selfAsInt
+        }
+        return Int(exactly: self)
+    }
+
+    /// Create an `Int32` from this `BinaryInteger`.
+    /// This will return nil if the conversion cannot result in an exact representation.
+    public func toInt32() -> Int32? {
+        if let selfAsInt32 = self as? Int32 {
+            return selfAsInt32
+        }
+        return Int32(exactly: self)
+    }
+
+    /// Create an `Int64` from this `BinaryInteger`.
+    /// This will return nil if the conversion cannot result in an exact representation.
+    public func toInt64() -> Int64? {
+        if let selfAsInt64 = self as? Int64 {
+            return selfAsInt64
+        }
+        return Int64(exactly: self)
+    }
+
+    /// Create a `Double` from this `BinaryInteger`.
+    /// This will return nil if the conversion cannot result in an exact representation.
+    public func toDouble() -> Double? {
+        return Double(exactly: self)
+    }
+}
+
+/// Default conformance to `BSONNumber` for `BinaryFloatingPoint`s.
+extension BSONNumber where Self: BinaryFloatingPoint {
+    /// Create an `Int` from this `BinaryFloatingPoint`.
+    /// This will return nil if the conversion cannot result in an exact representation.
+    public func toInt() -> Int? {
+        return Int(exactly: self)
+    }
+
+    /// Create an `Int32` from this `BinaryFloatingPoint`.
+    /// This will return nil if the conversion cannot result in an exact representation.
+    public func toInt32() -> Int32? {
+        return Int32(exactly: self)
+    }
+
+    /// Create an `Int64` from this `BinaryFloatingPoint`.
+    /// This will return nil if the conversion cannot result in an exact representation.
+    public func toInt64() -> Int64? {
+        return Int64(exactly: self)
+    }
+
+    /// Create a `Double` from this `BinaryFloatingPoint`.
+    /// This will return nil if the conversion cannot result in an exact representation.
+    public func toDouble() -> Double? {
+        if let selfAsDouble = self as? Double {
+            return selfAsDouble
+        }
+        return Double(self)
+    }
+}
+
+/// Default implementation of `Decimal128` conversions for all `Numeric`s.
+extension BSONNumber where Self: Numeric {
+    /// Create a `Decimal128` from this `Numeric`.
+    /// This will return nil if the conversion cannot result in an exact representation.
+    public func toDecimal128() -> Decimal128? {
+        return Decimal128(String(describing: self))
```

Comment:
how confident are we that the description strings will always be valid `Decimal128` strings? I seem to recall that the formatting of those is a little tricky. it might be worth testing a few more cases, like negative numbers.